src/solvers/refinement/string_refinement.cpp

```diff
@@ -923,7 +923,66 @@ void string_refinementt::substitute_array_access(exprt &expr) const
   }
 }
 
-/// negates the constraint and add it to the solver. the intended usage is to
+/// Negates the constraint and adds it to the solver. The intended usage is to
+/// find an assignement of the universal variable that would violate the axiom,
+/// to avoid false positives the symbols other than the universal variable
+/// should have been replaced by there valuation in the current model
+/// \param axiom: the not_contains constraint to add the negation of
+/// \param val: the existential witness for the axiom
+/// \param solver: the solver
+/// \return: univ_var: the valuation of the universal variable for the negation
+///   of the axiom.
+void string_refinementt::add_negation_of_not_contains_constraint_to_solver(
+  const string_not_contains_constraintt &axiom,
```

Comment:
I named this `axiom` (as it is in `string_refinementt::add_negation_of_constraint_to_solver`), and we do only call this function with things that have been put in `string_refinementt::axioms`. At this point, it doesn't need to be differentiated from another logical fragment, so I don't think it will be confused for anything else. If you feel like it is better to use `not_contains`, I can change it.